features/luks/data/lib/uevent/handlers/085-luks

```diff
@@ -48,8 +48,26 @@ match_dev_in_array() {
 	return 1
 }
 
+luks_unl0kr_ask_pass() {
+	[ "$#" = 2 ] || return 1
+
+	local cmd="$1" tries=$2 rc=2
+
+	plymouth hide-splash 2>/dev/null
+
+	while [ $rc != 0 ] && [ $tries -ne 0 ]; do
```

Comment:
The variables should be quoted.

Reply:
`rc` will always be a number, `tries` may be not a numbe,r but when it is `test` returns non-zero exit code

Reply:
> `rc` will always be a number, `tries` may be not a number, but when it is `test` returns non-zero exit code

Hm. In this case, this line looks even stranger. `!=` is condition in string context, `-ne` is condition in number context.

I would write it like this:
```shell
	while [ "$rc" != "0" ] && [ "$tries" != "0" ]; do
```
In this case, it doesn’t matter at all what the values ​​are in the variables.

Reply:
@legionus as [here](https://github.com/osboot/make-initrd/pull/39/files#diff-d50c296c234eaa6a3a14f88ba4dcda4fb99511e4197cfe5e174315a8f098921eR219)